nodejs/packages/layer/src/wrapper.ts

```diff
@@ -504,21 +512,27 @@ async function initializeLoggerProvider(
   );
 
   const logExporter = new OTLPLogExporter();
-  const logRecordProcessors: LogRecordProcessor[] = [];
-  const loggerConfig = {
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  let loggerConfig: any = {
     resource,
-    processors: logRecordProcessors,
+    processors: [],
   };
-  if (typeof configureLoggerProvider !== 'function') {
-    logRecordProcessors.push(new BatchLogRecordProcessor(logExporter));
+  if (typeof configureLogger === 'function') {
+    loggerConfig = configureLogger(loggerConfig);
+  }
+
+  loggerConfig.processors = loggerConfig.processors || [];
+  if (loggerConfig.processors.length === 0) {
+    loggerConfig.processors.push(new BatchLogRecordProcessor(logExporter));
   }
   // Logging for debug
   if (logLevel === DiagLogLevel.DEBUG) {
-    logRecordProcessors.push(
+    loggerConfig.processors.push(
       new SimpleLogRecordProcessor(new ConsoleLogRecordExporter()),
     );
   }
-  const loggerProvider = new LoggerProvider(loggerConfig);
+
+  const loggerProvider = new LoggerProvider(loggerConfig as object);
```

Comment:
Here we would no longer need to do the type assertion if we define the return type of `configureLogger` as suggested.

```suggestion
  const loggerProvider = new LoggerProvider(loggerConfig);
```